src/aind_metadata_service/sharepoint/client.py

```diff
@@ -0,0 +1,406 @@
+        END_OF_WEEK = "End_x0020_of_x0020_Week"
+        AGE_AT_INJECTION = "Age_x0020_at_x0020_Injection"
+        CRANIOTOMY_TYPE = "CraniotomyType"
+        IMPLANT_ID_COVERSLIP_TYPE = "ImplantIDCoverslipType"
+        INJ1_ANGLE_V2 = "Inj1Angle_v2"
+        INJ2_ANGLE_V2 = "Inj2Angle_v2"
+        FIBER_IMPLANT1 = "FiberImplant1"
+        FIBER_IMPLANT1_DV = "FiberImplant1DV"
+        FIBER_IMPLANT2 = "FiberImplant2"
+        FIBER_IMPLANT2_DV = "FiberImplant2DV"
+        ID2 = "ID"  # For some reason ID and Id are present in response
+        ODATA_UI_VERSION_STRING = "OData__UIVersionString"
+        ATTACHMENTS = "Attachments"
+        GUID = "GUID"
+
+
+class ListVersions(Enum):
+    """Enum class to handle different SharePoint list versions."""
+
+    VERSION_2019 = {
+        "list_title": "SWR 2019-Present",
+        "view_title": "New Request",
+    }
+    DEFAULT = {"list_title": "SWR 2019-Present", "view_title": "New Request"}
+
+
+class SharePointClient:
+    """This class contains the api to connect to SharePoint db."""
+
+    def __init__(
+        self, site_url: str, client_id: str, client_secret: str
+    ) -> None:
+        """
+        Initialize a client
+        Parameters
+        ----------
+        site_url : str
+           sharepoint site url
+        client_id : str
+            username for principal account to access sharepoint
+        client_secret : str
+            password for principal account to access sharepoint
+        """
+        self.site_url = site_url
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.credentials = ClientCredential(self.client_id, self.client_secret)
+        self.client_context = ClientContext(self.site_url).with_credentials(
+            self.credentials
+        )
+
+    @staticmethod
+    def _get_filter_string(version: ListVersions, subject_id: str) -> str:
+        """
+        Helper method to return a filter on the list
+        Parameters
+        ----------
+        version : ListVersions
+          Which version of the backend is being queried
+        subject_id : str
+          ID of the subject being queried for
+
+        Returns
+        -------
+        str
+          A string to pass into a query filter
+
+        """
+        default = (
+            f"substringof("
+            f"{subject_id}, "
+            f"{NeurosurgeryAndBehaviorList2019.ListField.LAB_TRACKS_ID.value}"
+            f")"
+        )
+        version_2019 = (
+            f"substringof("
+            f"{subject_id}, "
+            f"{NeurosurgeryAndBehaviorList2019.ListField.LAB_TRACKS_ID.value}"
+            f")"
+        )
+        # TODO: Handle other versions
+        filter_string = default
+        if version == ListVersions.VERSION_2019:
+            filter_string = version_2019
+        return filter_string
+
+    def get_procedure_info(
+        self,
+        subject_id: str,
+        version: ListVersions = ListVersions.VERSION_2019,
```

Comment:
probably best not to default to 2019, since we are going to have to look in two places all the time.